File: java/ATM.java
import java.util.Scanner;

public class ATM {
    
    public static boolean validasiNama(String nama) {
        return nama.matches("[a-zA-Z]+");
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String nama;

        do {
            System.out.print("Masukkan nama: ");
            nama = sc.nextLine();
            if (!validasiNama(nama)) {
                System.out.println("Nama tidak valid, coba lagi!");
            }
        } while (!validasiNama(nama));

        System.out.print("Masukkan NIM: ");
        String nim = sc.nextLine();
        long saldo = Long.parseLong(nim);

        System.out.println("\nHalo " + nama + ", saldo kamu Rp " + saldo);

        while (true) {
            System.out.println("\n===== MENU ATM =====");
            System.out.println("1. Cek Saldo");
            System.out.println("2. Tarik Tunai");
            System.out.println("3. Setor Tunai");
            System.out.println("4. Transfer");
            System.out.println("5. Keluar");
            System.out.print("Pilih menu: ");
            int pilih = sc.nextInt();

            switch (pilih) {
                case 1:
                    System.out.println("Saldo: Rp " + saldo);
                    break;

                case 2:
                    System.out.print("Jumlah tarik: ");
                    long tarik = sc.nextLong();
                    if (tarik <= saldo) saldo -= tarik;
                    else System.out.println("Saldo tidak cukup!");
                    break;

                case 3:
                    System.out.print("Jumlah setor: ");
                    long setor = sc.nextLong();
                    saldo += setor;
                    break;

                case 4:
                    sc.nextLine(); 
                    System.out.print("Rekening tujuan: ");
                    String rek = sc.nextLine();
                    System.out.print("Nominal transfer: ");
                    long nominal = sc.nextLong();

                    if (nominal <= saldo) saldo -= nominal;
                    else System.out.println("Saldo tidak cukup!");
                    break;

                case 5:
                    System.out.println("Terima kasih!");
                    return;

                default:
                    System.out.println("Menu tidak valid!");
            }
        }
    }
}

File: php/atm.php
<?php
function validasiNama($nama) {
    return preg_match("/^[a-zA-Z]+$/", $nama);
}

do {
    echo "Masukkan nama: ";
    $nama = trim(fgets(STDIN));

    if (!validasiNama($nama)) {
        echo "Nama tidak valid, coba lagi!\n";
    }
} while (!validasiNama($nama));

echo "Masukkan NIM: ";
$nim = trim(fgets(STDIN));

$saldo = intval($nim);
echo "\nHalo $nama, saldo kamu Rp " . number_format($saldo, 0, ',', '.') . "\n";

while (true) {
    echo "\n===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih menu: ";
    $pilih = trim(fgets(STDIN));

    switch ($pilih) {
        case 1:
            echo "Saldo: Rp " . number_format($saldo, 0, ',', '.') . "\n";
            break;

        case 2:
            echo "Jumlah tarik: ";
            $tarik = intval(fgets(STDIN));
            if ($tarik <= $saldo) {
                $saldo -= $tarik;
                echo "Penarikan berhasil!\n";
            } else {
                echo "Saldo tidak cukup!\n";
            }
            break;

        case 3:
            echo "Jumlah setor: ";
            $setor = intval(fgets(STDIN));
            $saldo += $setor;
            echo "Setoran berhasil!\n";
            break;

        case 4:
            echo "Rekening tujuan: ";
            $rek = trim(fgets(STDIN));
            echo "Nominal transfer: ";
            $nominal = intval(fgets(STDIN));

            if ($nominal <= $saldo) {
                $saldo -= $nominal;
                echo "Transfer berhasil ke $rek!\n";
            } else {
                echo "Saldo tidak cukup!\n";
            }
            break;

        case 5:
            echo "Terima kasih!\n";
            exit;

        default:
            echo "Menu tidak valid!\n";
    }
}
?>